packages/design-mui/src/components/MuiFilters/MuiFilters.tsx

```diff
@@ -55,6 +55,10 @@ export const MuiFilters: React.FC<XCollectionProps> = ({
     applyFilters(filters)
   }
 
+  useEffect(() => {
+    setFilters([defaultFilter])
+  }, [schemaName])
+
```

Comment:
fix this

Reply:
if the problem of infinite re-rendering is from putting `defaultFilter` in the deps array, have you tried using a useMemo for default filter?
```
defaultFilter = useMemo(() => ({...}), [fields])
```
_or_ since `defaultFilter` seems to depend on `fields`, we can update the `getFilterableFields` to return it?
```
const { fields, defaultFilter } = useMemo(() => getFilterableFields....
```
but i kinda like the first solution more

Reply:
I used a couple combinations of useCallback and useMemo, but not the right ones! Thanks for the insight. useMemo on defaultFilter and a useCallback on setFilter did the trick